### PR TITLE
pass URL params to filter results

### PR DESCRIPTION
**Pull request by stellaanaviW**
sample of filter usage to embedded app customers

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -40,7 +71,10 @@ export async function getServerSideProps() {
         }
     });
 
+    const searchTerms = ''; // 'searchTerm1,searchTerm2';
+    const filters = filtersMap['Tracking completed'];
```
Comment (ronfogel) [inline src/pages/index.tsx:75]: why?

File: src/pages/index.tsx [hunk 2]
```diff
@@ -10,7 +10,21 @@ const windwardEmbeddedDivStyle = {
...
+
+function getUrlParam(paramName:string, paramValue?:string) {
+    const param = paramValue? paramName + '=' + paramValue : undefined;
+    return param;
+}
+function Index({code, searchTerms, filters}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+
+    const searchTermsParam = getUrlParam(SEARCH_TERMS_URL_PARAM,searchTerms);
+    const filtersParam = getUrlParam(FILTERS_URL_PARAM,filters);
+    const urlParams = searchTermsParam && filtersParam ? searchTermsParam + AMPERSAND_URL + filtersParam : searchTermsParam || filtersParam;
```
Comment (ronfogel) [inline src/pages/index.tsx:26]: you can do that, and remove all the other related code

`const urlParams = new URLSearchParams({ filters, searchTerms }).toString();`

File: src/pages/index.tsx [hunk 3]
```diff
@@ -21,13 +35,30 @@ function Index({code}: InferGetServerSidePropsType<typeof getServerSideProps>) {
             {/* The following script must be placed below the div element.
                 This ensures that the fetched script can find and interact with the div element it relies on. */}
             <Script data-id={WINDWARD_EMBEDDED_SCRIPT_ID}
-                    src={`${process.env.NEXT_PUBLIC_EMBEDDED_APP_SCRIPT_ORIGIN}/embed.min.js`}></Script>
+                    src={`${process.env.NEXT_PUBLIC_EMBEDDED_APP_SCRIPT_ORIGIN}/embed.min.js${urlParams? QUESTION_MARK + urlParams: ''}`}></Script>
```
Comment (ronfogel) [inline src/pages/index.tsx:38]: it's OK to have question mark without anything after that.

```
<Script data-id={WINDWARD_EMBEDDED_SCRIPT_ID}
                    src={`${process.env.NEXT_PUBLIC_EMBEDDED_APP_SCRIPT_ORIGIN}/embed.min.js?${urlParams}`}></Script>
```

**Review by ronfogel — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx (hunk 1 above)
Comment (stellaanaviW) [reply src/pages/index.tsx:75]: by mistake, it should be commented out (probably left from my tests)

File: src/pages/index.tsx (hunk 2 above)
Comment (stellaanaviW) [reply src/pages/index.tsx:26]: cool

File: src/pages/index.tsx [hunk 4]
```diff
@@ -10,20 +10,11 @@ const windwardEmbeddedDivStyle = {
...
-function getUrlParam(paramName:string, paramValue?:string) {
-    const param = paramValue? paramName + '=' + paramValue : undefined;
-    return param;
-}
 function Index({code, searchTerms, filters}: InferGetServerSidePropsType<typeof getServerSideProps>) {
 
-    const searchTermsParam = getUrlParam(SEARCH_TERMS_URL_PARAM,searchTerms);
-    const filtersParam = getUrlParam(FILTERS_URL_PARAM,filters);
-    const urlParams = searchTermsParam && filtersParam ? searchTermsParam + AMPERSAND_URL + filtersParam : searchTermsParam || filtersParam;
+    const urlParams = filters && searchTerms ? new URLSearchParams({ filters, searchTerms})
```
Comment (ronfogel) [inline src/pages/index.tsx:15]: why all this mess?
`const urlParams = new URLSearchParams(omitBy({ filters, searchTerms }, isEmpty)).toString();`

the above should work. install `lodash` as dependency and probably `@types/lodash` as dev dependency for the typescript

File: src/pages/index.tsx [hunk 5]
```diff
@@ -40,7 +62,10 @@ export async function getServerSideProps() {
         }
     });
 
+    const searchTerms = ''; // 'searchTerm1,searchTerm2';
+    const filters = ''; // filtersMap['Tracking completed'];
```
Comment (ronfogel) [inline src/pages/index.tsx:66]: I would remove it, it's not needed.
Instead, I would use the `context` parameter from `getServerSideProps` to get the filters/searchTerms from the url itself and pass it to the component.

```
export async function getServerSideProps({ query }) {
...

const { searchTerms = '', filters = '' } = query ?? {};

return {
        props: {code, searchTerms, filters}
    }
```

this way, you can add the parameters to the url which will pass it down
http://localhost:3000/?searchTerms=something&....

**Review by ronfogel — CHANGES_REQUESTED** [review]

File: .env [hunk 6]
```diff
@@ -1,4 +1,4 @@
 NEXT_PUBLIC_GRAPHQL_GATEWAY_URL=https://graphql.wnwd.com
 NEXT_PUBLIC_EMBEDDED_APP_SCRIPT_ORIGIN=https://ofv.wnwd.com
-EMBEDDED_APP_SCRIPT_CLIENT_ID=clientId
-EMBEDDED_APP_SCRIPT_CLIENT_SECRET=clientSecret
+EMBEDDED_APP_SCRIPT_CLIENT_ID=Stella
+EMBEDDED_APP_SCRIPT_CLIENT_SECRET=2e15a83702ccc00bce09dce494e374cf3df2fc841188b4a8
```
[the remark this answers is not in the document]
Comment (stellaanaviW) [reply .env:4]: by mistake

File: src/pages/index.tsx (hunk 5 above)
Comment (stellaanaviW) [reply src/pages/index.tsx:66]: nice

**Review by ronfogel — APPROVED** [review]

**Pull request merged**